src/components/layout/RightSectionWallet.vue

```diff
@@ -21,25 +21,50 @@
         {{ address }}
       </div>
       <div class="not-bootstrap-row">
-        <AeAmount :amount="balance" />
+        <Dropdown
+          v-if="hasContractV2Address"
+          :options="tokenBalancesOptions"
+          :method="updateTokenBalance"
+          :selected="selectedTokenBalance.token"
+        >
+          <template #diplayValue>
+            <AeAmount
+              class="not-bootstrap-row"
+              :amount="selectedTokenBalance.balance"
+              :token="selectedTokenBalance.token"
+            />
+          </template>
+          <template slot-scope="{ option }">
+            <TokenAvatarAndSymbol :address="option.token" />
+          </template>
+        </Dropdown>
+        <AeAmount
+          v-else
+          :amount="balance"
+        />
         <Dropdown
           v-if="currencyDropdownOptions.length"
           :options="currencyDropdownOptions"
-          :method="updateCurrency"
+          :method="updateSelectedCurrency"
```

Comment:
```suggestion
          :method="({ currency }) => updateCurrency(currency)"
```
No need to wrap `updateCurrency` with an additional method for that